### ISSUE TITLE
xIisMimeTypeMapping: Make ConfigurationPath a non-mandatory parameter

### ISSUE THREAD

**Issue by sk82jack** [dsccommunity/xWebAdministration#577]
### Details of the scenario you tried and the problem that is occurring

I tried to use the `xIisMimeTypeMapping` resource without specifying the `ConfigurationPath` parameter as it wasn't listed as a property in the repository `readme.md` file. I then got an error saying that I was missing a mandatory parameter `ConfigurationPath`. After looking at the source code there is already a default value defined for this parameter if it's not specified [here](https://github.com/dsccommunity/xWebAdministration/blob/master/source/DSCResources/MSFT_xIisMimeTypeMapping/MSFT_xIisMimeTypeMapping.psm1#L10) so it's unnecessary for the parameter to be mandatory.

#### Verbose logs showing the problem

N/A

#### Suggested solution to the issue

In the [MSFT_xIisMimeTypeMapping.psm1](https://github.com/dsccommunity/xWebAdministration/blob/master/source/DSCResources/MSFT_xIisMimeTypeMapping/MSFT_xIisMimeTypeMapping.psm1) file change lines [35](https://github.com/dsccommunity/xWebAdministration/blob/master/source/DSCResources/MSFT_xIisMimeTypeMapping/MSFT_xIisMimeTypeMapping.psm1#L35), [107](https://github.com/dsccommunity/xWebAdministration/blob/master/source/DSCResources/MSFT_xIisMimeTypeMapping/MSFT_xIisMimeTypeMapping.psm1#L107) and [181](https://github.com/dsccommunity/xWebAdministration/blob/master/source/DSCResources/MSFT_xIisMimeTypeMapping/MSFT_xIisMimeTypeMapping.psm1#L181) from `[Parameter(Mandatory = $true)]` to `[Parameter()]` so that the parameter is not mandatory and will use the `MACHINE/WEBROOT/APPHOST` default value if not specified.

There should also be an update to the `readme.md` file to reference this parameter.

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
N/A

#### The operating system the target node is running
N/A

#### Version and build of PowerShell the target node is running
N/A

#### Version of the DSC module that was used
3.1.1

**Comment by johlju** [comment]
This property needs to be mandatory and a key in the schema, otherwise it wouldn't be possible to have this configuration.

```powershell
        xIisMimeTypeMapping Mpeg
        {
            Ensure            = 'Present'
            Extension         = '.mpeg'
            MimeType          = 'video/mpeg'
            ConfigurationPath = "IIS:\sites\Site1"
        }

        xIisMimeTypeMapping Mpeg
        {
            Ensure            = 'Present'
            Extension         = '.mpeg'
            MimeType          = 'video/mpeg'
            ConfigurationPath = "IIS:\sites\Site2"
        }
```

I think instead the default values (the constants) should be removed. 🤔

**Comment by sk82jack** [comment]
I'm not saying to remove the value or parameter as I know you need to pass something to the backend functions but making the parameter non-mandatory would allow you to pick up the default server-wide ConfigurationPath and still allow you to override using the config you posted. But yeah I guess you could go the other direction and keep it mandatory and then remove the unnecessary checks for a empty variables.

I'm not too sure what the best practices are in terms of DSC resources and default values (in normal modules it's widely accepted to set default values in the param blocks) but for me, I didn't know what I needed to pass in to the `ConfigurationPath` to add a mime type mapping at the server level until I checked the source code and saw that default value. I guess as long as it's documented somewhere or another example is added on how to add a mime type mapping at the server level then I'm not too fussed either way :)

**Comment by sk82jack** [comment]
Ah I just realised I glossed over your point `and a key in the schema`

I'm pretty new to DSC so I'm not too sure about how DSC schemas are implemented and used and didn't even look in the schema file until you mentioned it lol

So I guess those keys are what define the uniqueness of the DSC resources for want of a better phrase? In that case I understand why it would be a required parameter.

**Comment by johlju** [comment]
Yes exactly. You are correct that the properties that have the type Key define uniqueness. It cannot exist two resource instances with the same values for those parameters in the same configuration to be able to enforce the desired state. Mandatory parameter can for that reason not have default values.

There are four possible types of type qualifiers in the schema; Key, Required, Write, and Read.
For each schema property that has Key or Required the corresponding parameter in the PowerShell module script must mandatory. But only Key properties define uniqueness.

**Comment by DennisL68** [comment]
The resource xIisMimeTypeMapping has been renamed.
Bug fixes and new functionality will only be added to the renamed resource.
